libs/features/google-login.js

```diff
@@ -0,0 +1,53 @@
+const GOOGLE_SCRIPT = 'https://accounts.google.com/gsi/client';
+const GOOGLE_ID = '530526366930-l874a90ipfkn26naa71r010u8epp39jt.apps.googleusercontent.com';
+const PLACEHOLDER = 'feds-googleLogin';
+const WRAPPER = 'feds-profile';
+
+const onToken = async (getMetadata, data) => {
+  let destination;
+  try {
+    destination = new URL(getMetadata('google-login-redirect'))?.href;
+  } catch {
+    // Do nothing
+  }
+
+  await window.adobeIMS.socialHeadlessSignIn({
+    provider_id: 'google',
+    idp_token: data?.credential,
+    client_id: window.adobeid?.client_id,
+    scope: window.adobeid?.scope,
+  }).then(() => {
+    if (window.DISABLE_PAGE_RELOAD === true) return;
+    // Existing account
+    if (destination) {
+      window.location.assign(destination);
+    } else {
+      window.location.reload();
+    }
+  }).catch(() => {
+    // New account
+    window.adobeIMS.signInWithSocialProvider('google', { redirect_uri: destination || window.location.href });
+  });
+};
+
+export default async function initGoogleLogin(loadIms, getMetadata, loadScript) {
+  try {
+    await loadIms();
+  } catch {
+    return;
+  }
```

Comment:
Since you have the isSignedInUser check anyway, you could alternatively do this:
```suggestion
    await loadIms().catch(() => {})
```

Reply:
Oh you return if the user is already logged in, not NOT logged in. My bad! No in this case I think it's fine as is

Reply:
Also, I added one more check in `delayed` - https://github.com/adobecom/milo/pull/1157/files#diff-c21f6b181685a27721a54ce2b519823f6186157499b08ea290437d3fdff64473R55 - I wanted to make sure we do a fast fail and avoid loading the `google-login` module altogether if the user is logged-in already.